contractsdk/cpp/reserved/forbidden.cc

```diff
@@ -0,0 +1,47 @@
+#include "xchain/xchain.h"
+
+class Forbidden : public xchain::Contract {};
+
+DEFINE_METHOD(Forbidden, initialize) {
+    xchain::Context* ctx = self.context();
+    ctx->ok("initialize success");
+}
+
+DEFINE_METHOD(Forbidden, forbid) {
+    xchain::Context* ctx = self.context();
+    // txid to be forbidden
+    const std::string key = ctx->arg("txid");
+    // the reason to be forbidden
+    const std::string value = ctx->arg("value");
+    bool ret = ctx->put_object(key, value);
+    if (!ret) {
+        ctx->error("forbid failed");
+        return;
+    }
+    ctx->ok("forbid success");
+}
+
+DEFINE_METHOD(Forbidden, unforbid) {
+    xchain::Context* ctx = self.context();
+    // txid to be unforbidden
+    const std::string key = ctx->arg("txid");
+    bool ret = ctx->delete_object(key);
+    if (!ret) {
+        ctx->error("unforbid failed");
+        return;
+    }
+    ctx->ok("unforbid success");
+}
+
+DEFINE_METHOD(Forbidden, get) {
+    xchain::Context* ctx = self.context();
+    // check if txid has been forbidden
+    const std::string key = ctx->arg("txid");
+    std::string value;
+    bool ret = ctx->get_object(key, &value);
+    if (ret) {
+        ctx->ok("txid has been forbidden");
+        return;
+    }
+    ctx->error("txid has not been forbidden");
```

Comment:
error() will return using error code 500, so we indicate "not found" as "internal error"?

Reply:
Here,  "Key not found" is same as the message of "txid has not been forbidden" because of the contract of forbidden being blacklist.